### ISSUE TITLE
Question: Explore variants of nested enums without tracing them directly

### ISSUE THREAD

**Issue by rokob** [novifinancial/serde-reflection#98]
The README for the serde-reflection crate says:

> For efficiency reasons, the current algorithm does not attempt to scan the variants of enums other than the parameter T of the main call trace_type<T>. As a consequence, each enum type must be traced separately.

Is this really for efficiency reasons or is it not possible? It seems like one needs to call deserialize on a type that is not known at compile time to traverse  into the variants of an enum which is embedded inside T.

```
#[derive(Serialize, Deserialize)]
enum Example {
  Unit,
  Other(Other),
}

#[derive(Serialize, Deserialize)]
enum Other {
  A,
  B { a: u16 }
}
```

If I were to call only `trace_type::<Example>()` then I don't think it is possible to get the structure of `Other::B`. Or is there some way to do it that would be otherwise inefficient or potentially lead to an infinite loop?

Basically, is there a way to call `trace_type::<Example>()` and get all type information for all types contained therein? I know it is not possible in the current code, and I am not really asking for a change to this library to support it, but I'm curious if you might have an idea on how to do it?

**Comment by matbd** [comment]
No worries. Thanks for the comment!

**Comment by matbd** [comment]
See also #104
